### PR TITLE
feat(examples): with-grafbase

### PR DESCRIPTION
**Pull request by notrab**
This PR adds the `with-grafbase` example. `with-` prefix can be removed to follow guidelines but unsure if this is outdated since all other examples include this for similar services.

## Documentation / Examples

- [x] Make sure the linting passes by running `pnpm build && pnpm lint`
- [x] The "examples guidelines" are followed from [our contributing doc](https://github.com/vercel/next.js/blob/canary/contributing/examples/adding-examples.md)

## Todo
- [x] Vercel deploy button
- [x] Grafbase deploy button (with template schema)

### CODE REVIEW

File: examples/with-grafbase/.env.example [hunk 1]
```diff
@@ -0,0 +1,2 @@
+GRAFBASE_API_URL=http://localhost:4000/graphql
+GRAFBASE_API_KEY=
```
Comment (balazsorban44) [inline examples/with-grafbase/.env.example:2]: Assuming this is a secret, `.env.local.example` is more appropriate for a file name.

File: examples/with-grafbase/.eslintrc.json [hunk 2]
```diff
@@ -0,0 +1,3 @@
+{
+  "extends": "next/core-web-vitals"
```
Comment (balazsorban44) [inline examples/with-grafbase/.eslintrc.json:2]: Should be removed as per https://github.com/vercel/next.js/blob/canary/contributing/examples/adding-examples.md

File: examples/with-grafbase/package.json [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    "start": "next start",
+    "lint": "next lint",
+    "codegen": "graphql-codegen --watch -r dotenv/config"
+  },
+  "dependencies": {
+    "@types/node": "18.11.9",
+    "@types/react": "18.0.25",
+    "@types/react-dom": "18.0.8",
+    "eslint": "8.27.0",
+    "eslint-config-next": "13.0.3",
```
Comment (balazsorban44) [inline examples/with-grafbase/package.json:18]:
```suggestion
```

File: examples/with-grafbase/package.json [hunk 4]
```diff
@@ -0,0 +1,34 @@
+{
+  "name": "with-grafbase",
+  "version": "0.1.0",
+  "private": true,
+  "scripts": {
+    "dev": "next dev",
+    "backend": "npx grafbase@latest dev & yarn codegen",
+    "build": "next build",
+    "start": "next start",
+    "lint": "next lint",
```
Comment (balazsorban44) [inline examples/with-grafbase/package.json:10]:
```suggestion
```

File: examples/with-grafbase/package.json [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+  },
+  "dependencies": {
+    "@types/node": "18.11.9",
+    "@types/react": "18.0.25",
+    "@types/react-dom": "18.0.8",
+    "eslint": "8.27.0",
+    "eslint-config-next": "13.0.3",
+    "graphql": "16.6.0",
+    "graphql-request": "5.0.0",
+    "next": "13.0.3",
```
Comment (balazsorban44) [inline examples/with-grafbase/package.json:21]:
```suggestion
    "next": "latest",
```

**Comment by notrab** [comment]
Thanks for reviewing @balazsorban44. I didn't quite finish pushing changes yesterday hence the draft PR so most of your comments should be resolved in that.

**Review by balazsorban44 — CHANGES_REQUESTED** [review]
Can you make sure that linting passes? https://github.com/vercel/next.js/actions/runs/3469499938/jobs/5796602833

you can run `pnpm prettier --write examples/with-grafbase` and commit the changed files

File: examples/with-grafbase/package.json [hunk 6]
```diff
@@ -0,0 +1,31 @@
+{
+  "name": "with-grafbase",
+  "version": "0.1.0",
```
Comment (balazsorban44) [inline examples/with-grafbase/package.json:3]:
```suggestion
```

**Comment by notrab** [comment]
@balazsorban44 all done 🙌🏻

**Review by balazsorban44 — APPROVED** [review]

**Comment by balazsorban44** [comment]
Thanks!

**Pull request merged**